### ISSUE TITLE
Measurement when doing frequency hops

### ISSUE THREAD

**Issue by xmikos** [AD-Vega/rtl-power-fftw#4]
For long sweeps with multiple frequency hops, `rtl_power_fftw` is now waiting (collecting samples) on each hop for time `--time` before going to next hop. Would it be possible to add mode in which it would sweep over whole range (over and over again) and do averaging only after collecting needed amount of samples from whole range?

Simply instead of doing (pseudocode):

    for hop in hops:
        for repeat in repeats:
            collect_samples()
        average_collected_samples()

Do this (pseudocode):

    for repeat in repeats:
        for hop in hops:
            collect_samples()
    average_collected_samples()

I know this mode would be slower (because of much more hops), but it would be better for spectrum analyzer usecase.

EDIT: Well, for some spectrum analyzer usecases, not all ;-)

**Comment by xmikos** [comment]
Now that I think about it, I can implement this mode by myself outside of `rtl_power_fftw` by using only 1 repeat in rtl_power_fftw (in continuous mode) and doing averaging externaly in my app. But I am not sure if this wouldn't be inefficient and much slower than doing it directly inside `rtl_power_fftw`?

**Comment by KlemenBlokar** [comment]
Frequency hopping is quite inefficient, and the same goes for measuring only one repeat at once. The latter is apparently due to the specifics of the USB protocol. This is why it would be better than measuring a single repeat to measure as many as needed to cover 16384 samples (one full transfer buffer) - or a small multiple of this, of course.
I think that considering the time you have to spare while waiting for another sweep to finish, you can easily average data within your app.
If you would require raw output for this (not logarithmic ), it would be easy to implement.

Redoing the loop in the more inefficient way would however require quite a rewrite of the logic, as I only need one buffer for averaging the power spectrum now - to implement the other idea I would need as many buffers as there are frequency hops and I would also need to remember all the tuned frequencies, etc... And on the downside, you would have to wait even longer to get data from each sweep, as now you get each hop immediately after its done, but then you would need to wait for the whole set to finish. So doing shorter sweeps and averaging them later yourself seems a much better option to me. But if you find there are problems with this way of doing things, we can think again. :-)

**Comment by xmikos** [comment]
You are probably right, I will try this way in QSpectrumAnalyzer to see how good it will work.

**Comment by KlemenBlokar** [comment]
Ok, I will label this as `wontfix` for the time being. :-)
